webrtc.html

```diff
@@ -8723,10 +8723,39 @@ <h3><dfn>RTCDataChannel</dfn></h3>
       <p>An <code><a>RTCDataChannel</a></code> object's <a>underlying data
       transport</a> may be torn down in a non-abrupt manner by running the
       <dfn id="data-transport-closing-procedure">closing procedure</dfn>. When
-      that happens the user agent MUST, unless the procedure was initiated by
-      the <code><a data-link-for="RTCDataChannel">close</a></code> method, queue a
-      task that sets the object's <a>[[\ReadyState]]</a> slot to <code>closing</code>.
-      This will eventually render the <a>data transport</a> <a>closed</a>.</p>
+      that happens the user agent MUST queue a task to run the following
+      steps:</p>
+      <ol>
+        <li>
+          <p>Let <var>channel</var> be the <code><a>RTCDataChannel</a></code>
+          object whose <a data-lt="data transport">transport</a> was
+          closed.</p>
+        </li>
+        <li>
+          <p>Unless the procedure was initiated by the <var>channel</var>'s
+          <code><a data-link-for="RTCDataChannel">close</a></code> method, set
+          <var>channel</var>'s <a>[[\ReadyState]]</a> slot to
+          <code>closing</code>.</p>
+        </li>
```

Comment:
I think the state update is the only operation that needs to queued on the main thread. A simple fix for that would be to bring back the text above that conditionally queued the task to fire the event, and then just "run these steps" for the rest.

Reply:
The task updating the state should fire the event at channel as well.

Reply:
> The task updating the state should fire the event at channel as well.

IIRC that would clash with the rule to not fire events due to user action (even though I don't like that rule).

Reply:
You're right that there's no event.